### PR TITLE
doc: improve some of str examples

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libcollections/str.rs [hunk 1]
```diff
@@ -1195,6 +1177,7 @@ impl str {
     ///
     /// ```
     /// assert_eq!("11foo1bar11".trim_matches('1'), "foo1bar");
+    /// assert_eq!("123foo1bar123".trim_matches(|c: char| c.is_numeric()), "foo1bar");
```
Comment (bluss) [inline src/libcollections/str.rs]: this could be `char::is_numeric` too?, just like the split example?

**Comment by bluss** [comment]
I like the new formulations, it should be friendlier to read. r=me with the nit fixed for consistency

File: src/libcollections/str.rs [hunk 2]
```diff
@@ -1494,7 +1472,7 @@ impl str {
     /// ```
     /// let s = "Löwe 老虎 Léopard";
     ///
-    /// assert_eq!(s.find(|c: char| c.is_whitespace()), Some(5));
+    /// assert_eq!(s.find(cahr::is_whitespace), Some(5));
```
Comment (bluss) [inline src/libcollections/str.rs]: small typo

Comment (tshepang) [inline src/libcollections/str.rs]: Sorry, it's fixed now. I meant only to ping you once the tests passed.

You may have noticed that this is one example of removing some more needless usage of closures.

**Comment by bluss** [comment]
Thank you!

**Pull request merged**